controllers/clusterurlmonitor/clusterurlmonitor_supplement.go

```diff
@@ -67,6 +68,16 @@ func (s *ClusterUrlMonitorReconciler) EnsurePrometheusRuleExists(clusterUrlMonit
 	return utilreconcile.ContinueReconcile()
 }
 
+// FindStatusCondition finds the conditionType in conditions.
+func (s *ClusterUrlMonitorReconciler) FindStatusCondition(conditions []metav1.Condition, conditionType string) *metav1.Condition {
```

Comment:
Are there any benefits to have our own implementation of FindStatusCondition as opposed to using https://pkg.go.dev/k8s.io/apimachinery/pkg/api/meta#FindStatusCondition?